pkg/installer/generateconfig.go

```diff
@@ -113,6 +114,15 @@ func (m *manager) generateInstallConfig(ctx context.Context) (*installconfig.Ins
 		outboundType = azuretypes.UserDefinedRoutingOutboundType
 	}
 
+	masterDiskEncryptionSet, err := azure.ParseResourceID(m.oc.Properties.MasterProfile.DiskEncryptionSetID)
+	if err != nil {
+		return nil, nil, err
+	}
+	workerDiskEncryptionSet, err := azure.ParseResourceID(m.oc.Properties.WorkerProfiles[0].DiskEncryptionSetID)
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
The call to ParseResourceID fails in the default casze when user encryption keys are not provided. I reworked this in the version that I tested successfully on 4.10 and 4.11 with the following changes:

```suggestion
	var masterDiskEncryptionSet *azuretypes.DiskEncryptionSet
	var workerDiskEncryptionSet *azuretypes.DiskEncryptionSet

	if m.oc.Properties.MasterProfile.DiskEncryptionSetID != "" {
		masterDiskEncryptionSetResource, err := azure.ParseResourceID(m.oc.Properties.MasterProfile.DiskEncryptionSetID)
		if err != nil {
			return nil, nil, err
		}

		masterDiskEncryptionSet = &azuretypes.DiskEncryptionSet{
			SubscriptionID: masterDiskEncryptionSetResource.SubscriptionID,
			ResourceGroup:  masterDiskEncryptionSetResource.ResourceGroup,
			Name:           masterDiskEncryptionSetResource.ResourceName,
		}
	}

	if m.oc.Properties.WorkerProfiles[0].DiskEncryptionSetID != "" {
		workerDiskEncryptionSetResource, err := azure.ParseResourceID(m.oc.Properties.WorkerProfiles[0].DiskEncryptionSetID)
		if err != nil {
			return nil, nil, err
		}

		workerDiskEncryptionSet = &azuretypes.DiskEncryptionSet{
			SubscriptionID: workerDiskEncryptionSetResource.SubscriptionID,
			ResourceGroup:  workerDiskEncryptionSetResource.ResourceGroup,
			Name:           workerDiskEncryptionSetResource.ResourceName,
		}
	}
```